gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/internal/user/UserMongoRepository.java

```diff
@@ -34,9 +34,15 @@ public interface UserMongoRepository extends MongoRepository<UserMongo, String>,
     Set<UserMongo> findByIds(Collection<String> ids);
 
     @Query(value = "{ 'source': ?0, 'sourceId': {$regex: '^?1$', $options: 'i'}, 'organizationId': ?2 }")
-    UserMongo findBySourceAndSourceId(String source, String sourceId, String organizationId);
+    UserMongo findBySourceAndSourceIdIgnoreCase(String source, String sourceId, String organizationId);
 
     @Query(value = "{ 'email': {$regex: '^?0$', $options: 'i'}, 'organizationId': ?1 }")
+    UserMongo findByEmailIgnoreCase(String email, String organizationId);
+
+    @Query(value = "{ 'source': ?0, 'sourceId': ?1, 'organizationId': ?2 }")
+    UserMongo findBySourceAndSourceId(String source, String sourceId, String organizationId);
+
+    @Query(value = "{ 'email':?0, 'organizationId': ?1 }")
```

Comment:
Juste a remind : 
I don't know if in the database the email or sourceId have a value with capital letters in the database. If this is the case, they will need to be passed in lowercase with the encprition of the data. and remember to test it.
right ?

Reply:
ToLowercase has been added in the migration script and successfully tested